test/format/config.c

```diff
@@ -715,51 +716,48 @@ static void
 config_directio(void)
 {
     /*
-     * We don't roll the dice and set direct I/O, it has to be set explicitly. For that reason, any
-     * incompatible "permanent" option set with direct I/O is a configuration error.
+     * We don't roll the dice and set direct I/O, it has to be set explicitly. If there are any
+     * incompatible configurations set explicitly, turn off direct I/O, otherwise turn off the
+     * incompatible configurations.
      */
     if (!GV(DISK_DIRECT_IO))
         return;
+    testutil_assert(config_explicit(NULL, "disk.direct_io") == true);
+
+#undef DIO_CHECK
+#define DIO_CHECK(name, flag)                                                       \
+    if (GV(flag)) {                                                                 \
+        if (config_explicit(NULL, name)) {                                          \
+            WARN("%s not supported with direct I/O, turning off direct I/O", name); \
+            config_single(NULL, "disk.direct_io=off", false);                       \
+            return;                                                                 \
+        }                                                                           \
+        config_single(NULL, #name "=off", false);                                   \
+    }
 
     /*
      * Direct I/O may not work with backups, doing copies through the buffer cache after configuring
      * direct I/O in Linux won't work. If direct I/O is configured, turn off backups.
      */
-    if (GV(BACKUP)) {
-        if (config_explicit(NULL, "backup"))
-            testutil_die(EINVAL, "direct I/O is incompatible with backup configurations");
-        config_single(NULL, "backup=off", false);
-    }
+    DIO_CHECK("disk.backup", BACKUP);
```

Comment:
believe we don't have disk.backup, and it's called `backup`.

Reply:
Thank you, @jiechenbo, fixed in 246503404.